src/ratbag_emu/server/device_controller.py

```diff
@@ -168,13 +166,13 @@ def get_led(device_id, led_id):
        DeviceHandler.devices[device_id] is None:
```

Comment:
comment on the length check here: general attitude in python is to use exceptions for stuff that shouldn't happen. So more pythonic would be something like this:
```
try:
  device =  DeviceHandler.devices[device_id]
except IndexError:
  device = None
finally:
  if device is None:
    return ...
```

Reply:
also, I'm a bit confused here (because it's getting late). The `DeviceHandler.devices` is a list but you're using a device ID as index. Which means you need to a) guarantee that the index increments from 0 and b) never remove any device from the list. This seems like it would be better handled by a dictionary.

Reply:
`DeviceHandler.devices` is a dict.

https://github.com/libratbag/ratbag-emu/blob/07fdaf5b22354798771b20b6263c45be468d824e/src/ratbag_emu/device_handler.py#L11

Reply:
Then you should have a class, and not a dict

Reply:
oops, sorry, I thought you were referring to my comment, ignore that (I haven't gave thought at this part, yet)